src/plugins/ui_stateful/selection.ts

```diff
@@ -420,17 +421,39 @@ export class GridSelectionPlugin extends UIPlugin {
 
   getStatisticFnResults(): { [name: string]: number | undefined } {
```

Comment:
Remi optimized this funciton in his [task](https://www.odoo.com/web#id=3827324&menu_id=4720&cids=1&action=333&active_id=2328&model=project.task&view_type=form)

His branch contains the `isRowHiddenCache` optimization.

I'd rebase your branch on his to see if your `recomputedZones` optimization has an practical impact in performances and is useful.